tests/MongoDB.Driver.Examples/TransactionExamplesForDocs/WithTransactionExample1.cs

```diff
@@ -0,0 +1,100 @@
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using FluentAssertions;
+using MongoDB.Bson;
+using MongoDB.Driver.Core.Clusters;
+using MongoDB.Driver.Core.Misc;
+using MongoDB.Driver.Core.TestHelpers.XunitExtensions;
+using Xunit;
+
+namespace MongoDB.Driver.Examples.TransactionExamplesForDocs
+{
+    public class WithTransactionExample1
+    {
+        [SkippableFact]
+        public void Example1()
+        {
+            RequireServer.Check().ClusterTypes(ClusterType.ReplicaSet, ClusterType.Sharded).Supports(Feature.Transactions);
+
+            var connectionString = CoreTestConfiguration.ConnectionString.ToString();
+            DropCollections(
+                connectionString, 
+                CollectionNamespace.FromFullName("mydb1.foo"), 
+                CollectionNamespace.FromFullName("mydb2.bar"));
+            string result = null;
+
+            // Start Transactions withTxn API Example 1
+            // For a replica set, include the replica set name and a seedlist of the members in the URI string; e.g.
+            // string uri = "mongodb://mongodb0.example.com:27017,mongodb1.example.com:27017/?replicaSet=myRepl";
+            // For a sharded cluster, connect to the mongos instances; e.g.
+            // string uri = "mongodb://mongos0.example.com:27017,mongos1.example.com:27017:27017/";
+            var client = new MongoClient(connectionString);
+
+            // Prereq: Create collections. CRUD operations in transactions must be on existing collections.
+            var database1 = client.GetDatabase("mydb1");
+            var collection1 = database1.GetCollection<BsonDocument>("foo").WithWriteConcern(WriteConcern.WMajority);
+            collection1.InsertOne(new BsonDocument("abc", 0));
+
+            var database2 = client.GetDatabase("mydb2");
+            var collection2 = database2.GetCollection<BsonDocument>("bar").WithWriteConcern(WriteConcern.WMajority);
+            collection2.InsertOne(new BsonDocument("xyz", 0));
+
+            // Step 1: Start a client session.
+            using (var session = client.StartSession())
+            {
+                // Step 2: Optional. Define options to use for the transaction.
+                var transactionOptions = new TransactionOptions(
+                    readPreference: ReadPreference.Primary,
+                    readConcern: ReadConcern.Local,
+                    writeConcern: WriteConcern.WMajority);
+
+                // Step 3: Define the sequence of operations to perform inside the transactions
+                result = session.WithTransaction(
+                    (s, cancellationToken) =>
+                    {
+                        collection1.InsertOne(s, new BsonDocument("abc", 1), cancellationToken: cancellationToken);
+                        collection2.InsertOne(s, new BsonDocument("xyz", 999), cancellationToken: cancellationToken);
```

Comment:
Note: we actually have two session variables available to use here:

1. The captured `session` variable
2. The `s` parameter

Might as well use the captured session variable to be consistent with the surrounding code.

Reply:
This comment was made before.

I can't tell if you just missed it or if you specifically wanted to use `s` instead of `session`.

That would probably be OK, but I'd like to know the reason.

Reply:
Since we provide a session as a parameter for callback (`s`), I guess it's a good idea to use the provided parameter inside the callback delegate. So, I intentionally used this parameter.